zfs-replicate.sh

```diff
@@ -310,6 +310,46 @@ snapCreate() {
     ## get source and destination snapshots
     srcSnaps=$(snapList "$src" "$srcHost" 1)
     dstSnaps=$(snapList "$dst" "$dstHost" 0)
+    srcSnapsAll=$(snapList "$src" "$srcHost" 0)
+    dstSnapsAll=$(snapList "$dst" "$dstHost" 0)
+    ## check that all datasets have matching snapshots
+    ## reset fail variable
+    snapCheckFail=""
+    for ssnap in $srcSnapsAll; do
+      ## reset snapMatch variable
+      snapMatch=""
```

Comment:
This is used as a bool, so let's make it a 0 instead of empty and then check against 0 or 1.